openedx/core/lib/edx_api_utils.py

```diff
@@ -89,18 +90,15 @@ def get_edx_api_data(api_config, user, resource,
     return results
 
 
-def _traverse_pagination(response, endpoint, querystring, no_data):
-    """Traverse a paginated API response.
-
+def _traverse_pagination(response, endpoint, no_data):
+    """
+    Traverse a paginated API response.
     Extracts and concatenates "results" (list of dict) returned by DRF-powered APIs.
     """
     results = response.get('results', no_data)
-
-    page = 1
     next_page = response.get('next')
     while next_page:
-        page += 1
-        querystring['page'] = page
+        querystring = urlparse.parse_qs(urlparse.urlparse(next_page).query)
```

Comment:
This function was only handling page-number based DRF pagination and it undergoes in an infinite loop for any other style of pagination e.g limit-offset based DRF pagination.
Hence, we are only retrieving the query string from next result set url (retrieved in response of API) and calling it again to get next result set unless all results are retrieved instead of incrementing page value which won't be effective for limit-offset style pagination.